src/app/components/graph/graph-launcher.html

```diff
@@ -173,6 +173,12 @@ <h6>
                                         <ul
                                             class="dropdown-menu"
                                             ng-show="isVisible('tags')">
+                                            <li ng-click="onItemSelect('tags', 'select_all', $event)">
```

Comment:
by treating this as a regular item that can be selected, we can find ourselves in some funky states! Check this example out, for instance:

<img width="220" alt="Screen Shot 2020-09-25 at 2 00 47 PM" src="https://user-images.githubusercontent.com/1541139/94300863-979aa500-ff37-11ea-82f8-f893d036b00a.png">

Instead, I think we might just want to add a new method in the scope called something like `toggleSelectAll('tags')`. This method would either unselect all tags, or make all tags selected depending on the state. I think that would help clean up some of the logic added in `graph-launcher.js ` too!

Let me know if you have any thoughts or questions about an approach like this one :)

Reply:
I definitely see your point, I can change it to a separate function. That is what I had considered at first, I thought the current was just less intrusive.